rkyv/src/vec/mod.rs

```diff
@@ -178,6 +178,160 @@ impl<T> ArchivedVec<T> {
             Ok(VecResolver { pos })
         }
     }
+
+    /// Serializes an archived `Vec` from a given iterator. Compared to `serialize_from_iter()`,
+    /// this function:
+    /// - supports iterators whose length is not known in advance, and
+    /// - does not collect the data in memory before serializing.
+    ///
+    /// This method will panic if any item writes during `serialize` (i.e no additional data
+    /// written per item).
+    ///
+    /// Usage example:
+    /// ```
+    /// use rkyv::{vec::ArchivedVec, archived_root,
+    ///     ser::{serializers::{WriteSerializer, AllocSerializer}, Serializer},
+    ///     AlignedBytes, Archive, Deserialize, Serialize,
+    /// };
+    ///
+    /// const SCRATCH_SIZE: usize = 256;
+    /// type DefaultSerializer = AllocSerializer<SCRATCH_SIZE>;
+    ///
+    /// // Build some example data structures that are a little bit complex
+    /// #[derive(Clone, Copy, Debug, PartialEq, Archive, Serialize, Deserialize)]
+    /// #[archive(compare(PartialEq))]
+    /// #[archive_attr(derive(Clone, Copy, Debug))]
+    /// enum ExampleEnum {
+    ///     Foo,
+    ///     Bar(u64),
+    /// }
+    /// #[derive(Clone, Copy, Debug, PartialEq, Archive, Serialize, Deserialize)]
+    /// #[archive(compare(PartialEq))]
+    /// #[archive_attr(derive(Clone, Copy, Debug))]
+    /// struct Example {
+    ///     x: i32,
+    ///     y: ExampleEnum,
+    /// }
+    ///
+    /// // Build the data to stand-in for an Iterator. We assume that in practice, this underlying
+    /// // Vec will not be visible to the user: perhaps the data is being generated with every call
+    /// // iterator, or perhaps it is too large to fit in memory
+    /// let example_data = vec![
+    ///     Example {
+    ///         x: -5,
+    ///         y: ExampleEnum::Bar(42),
+    ///     },
+    ///     Example {
+    ///         x: -3,
+    ///         y: ExampleEnum::Foo,
+    ///     },
+    /// ];
+    ///
+    /// // If the Iterator length is not known, we have to wrap it in an adapter that tells us the length
+    /// // after the fact
+    /// struct IterCount<I, T>
```

Comment:
```suggestion
    /// struct IterCount<I>
```

Reply:
Is this a feature in a newer version of rustc? On 1.66.0 I get:
```
$ cargo test --doc serialize_from_unknown
    Finished test [unoptimized + debuginfo] target(s) in 0.04s
   Doc-tests rkyv

running 1 test
test src/vec/mod.rs - vec::ArchivedVec<T>::serialize_from_unknown_length_iter (line 191) ... FAILED

failures:

---- src/vec/mod.rs - vec::ArchivedVec<T>::serialize_from_unknown_length_iter (line 191) stdout ----
error[E0412]: cannot find type `T` in this scope
  --> src/vec/mod.rs:235:5
   |
44 | struct IterCount<I>
   |                  - similarly named type parameter `I` defined here
...
47 |     T: Copy,
   |     ^
   |
help: a type parameter with a similar name exists
   |
47 |     I: Copy,
   |     ~
help: you might be missing a type parameter
   |
44 | struct IterCount<I, T>
   |                   +++

error[E0282]: type annotations needed
  --> src/vec/mod.rs:237:11
   |
49 |     iter: I,
   |           ^ cannot infer type for type parameter `I`

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0282, E0412.
For more information about an error, try `rustc --explain E0282`.
Couldn't compile the test.

failures:
    src/vec/mod.rs - vec::ArchivedVec<T>::serialize_from_unknown_length_iter (line 191)

```

Reply:
Fixed in newest commit